students/WooseokJ/Lesson 8/CircleWJ.py

```diff
@@ -0,0 +1,151 @@
+import io
+import pytest
+import math
+from math import pi
+
+# Basic properties of circle
+class Circle():
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        return pi * (self.radius**2)
+
+#  alternator constructor that will let the use create a circle directly with the diameter
+    @classmethod
+    def from_diameter(cls, value):
+        return cls(value/2)
+
+    def __str__(self):
+        return "circle with radius: {:.4f}".format(self.radius)
+
+    def __repr__(self):
+        return "circle({})".format(self.radius)
+
+    def __add__(self, other):
+        return self.radius + other.radius

```

Comment:
This should return a Circle object that has the combined radius of the two circles added, not just the sum of the two radius.

Reply:
The same applies to your implementation of _rmul_, _mul_ and _div_. They should all return new Circle objects with the appropriate radius.